`make clean && make` is broken on master

**Describe the bug**
Make builds are broken after `make clean` on master

**To Reproduce**
start with a fresh clean checkout

```
$ git show
commit 7bc521a0a401c6c7ae8f6fe766ebfcc6d6e4fa01 (HEAD -> master, origin/master, origin/HEAD)
Merge: f403b34a6c 0a720d168e
Author: Ingela Anderton Andin <ingela@erlang.org>
Date:   Tue Apr 18 15:10:38 2023 +0200

$ ./configure
....

$ make -j
...
Fixing up development runtime dependencies # no errors

$ make clean 
... # no errors

$ make -j
...
 ERLC	../ebin/beam_asm.beam
beam_asm.erl:31:10: can't find include file "beam_opcodes.hrl"
%   31| -include("beam_opcodes.hrl").
%     |          ^

beam_asm.erl:316:14: undefined macro 'tag_i'
%  316|     [encode(?tag_i, L)|encode_line_items(T, F)];

make[3]: *** [Makefile:180: ../ebin/beam_asm.beam] Error 1
make[3]: Leaving directory '/home/ilya/code/otp/lib/compiler/src'
make[2]: *** [/home/ilya/code/otp/make/otp_subdir.mk:29: opt] Error 2
make[2]: Leaving directory '/home/ilya/code/otp/lib/compiler'
make[1]: *** [/home/ilya/code/otp/make/otp_subdir.mk:29: opt] Error 2
make[1]: Leaving directory '/home/ilya/code/otp/lib'
make: *** [Makefile:518: libs] Error 2

```

**Expected behavior**
`make clean && make` should be working (at it was before) 

**Affected versions**
OTP master (7bc521a0a401c6c7ae8f6fe766ebfcc6d6e4fa01)

**Additional context**

The breakage makes using CLion for development of OTP painful - as CLion may perform `make clean` to read the project structure (needed for intellisense)  - which literally breaks the development environment.

It's worth to note that it was not the case before (though I did not bisect to find the culprit change)

This is reproducible both on Linux and Mac.

--- a/lib/compiler/src/Makefile
+++ b/lib/compiler/src/Makefile
@@ -157,7 +157,7 @@ docs:
 
 clean:
 	rm -f $(TARGET_FILES)
-	rm -f $(EGEN)/beam_opcodes.erl $(EGEN)/beam_opcodes.hrl
+	rm -f $(EGEN)/beam_opcodes.erl $(EGEN)/beam_opcodes.hrl $(EGEN)/OPCODES-GENERATED
 	rm -f $(EGEN)/core_parse.erl
 	rm -f core
 

